pkg/releaseutil/manifest.go

```diff
@@ -37,12 +38,12 @@ func SplitManifests(bigfile string) map[string]string {
 	// Basically, we're quickly splitting a stream of YAML documents into an
 	// array of YAML docs. In the current implementation, the file name is just
 	// a place holder, and doesn't have any further meaning.
-	sep := "\n---\n"
+	sep := regexp.MustCompile("(?:^|\\s*\n)---\\s*")
 	tpl := "manifest-%d"
 	res := map[string]string{}
-	// Making sure yaml formatting doesn't matter when generating manifest from string.
+	// Making sure YAML formatting doesn't matter when generating manifest from string.
```

Comment:
- [ ] [As suggested earlier](https://github.com/kubernetes/helm/pull/2190#discussion_r108066539), consider revising this comment, as it doesn't read properly in American English.  
  > Make sure that any extra whitespace in the YAML stream doesn't preclude splitting its documents.